lib/waterfall.js

```diff
@@ -68,24 +68,23 @@ export default  function(tasks, callback) {
     if (!isArray(tasks)) return callback(new Error('First argument to waterfall must be an array of functions'));
     if (!tasks.length) return callback();
     var taskIndex = 0;
+    var args = [];
```

Comment:
Instead of creating this variable I would consider modifying `nextTask` to take args as a param (`nextTasks(args)`). In `next`, we would just call `nextTask(slice(arguments, 1))`